Userland/Libraries/LibWeb/WebAssembly/Table.cpp

```diff
@@ -106,8 +106,6 @@ WebIDL::ExceptionOr<JS::Value> Table::get(u32 index) const
         return vm.throw_completion<JS::RangeError>("Table element index out of range"sv);
 
     auto& ref = table->elements()[index];
-    if (!ref.ref().has<Wasm::Reference::Null>())
-        return JS::js_undefined();
 
     Wasm::Value wasm_value { ref };
     return Detail::to_js_value(vm, wasm_value);
```

Comment:
This function can't correctly handle null refs, the type given by .type() will be the deref'd type.

Let's keep the if..., but actually fix the bug:
```suggestion
    if (ref.ref().has<Wasm::Reference::Null>())
        return JS::js_null();
```

Reply:
Isn't to_js_value supposed to return null on a null ref?

Reply:
I don't think so?
it doesn't handle null refs at all.

Reply:
I'm pretty sure it should: https://webassembly.github.io/spec/js-api/#tojsvalue

> 6. If w is of the form [ref.null](https://webassembly.github.io/spec/core/exec/runtime.html#values) t, return null.

I'll include that in this PR, then.

Reply:
Please do, thanks!